### PR TITLE
[Feature] Signal handling change event

### PR DESCRIPTION
**Pull request by AlonZivony**
## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

Created a new event for the case that a user change signal handler (aka sigaction operation).

Fixes: #2185 

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

**Review by rafaeldtinoco — APPROVED** [review]
LGTM, event "confined to its own purpose" and "well documented". @AlonZivony non CO-RE seems problematic though. Could you fix and re-submit for my reviewal ? Thanks

File: pkg/ebpf/c/tracee.bpf.c
Comment (NDStrahilevitz) [inline pkg/ebpf/c/tracee.bpf.c:1699]: Is this duplicate on purpose?

Comment (rafaeldtinoco) [reply pkg/ebpf/c/tracee.bpf.c:1699]: Nice catch, I believe it is a mistake. We could have it fixed when merging through github UI but then tests don't run when we do manual changes, so It is better to have it fixed and code repushed.

Comment (AlonZivony) [reply pkg/ebpf/c/tracee.bpf.c:1699]: Wow, can't believe I didn't notice this.
I fix it and push the new version.

**Comment by AlonZivony** [comment]
@NDStrahilevitz can we merge it?

**Review by NDStrahilevitz — APPROVED** [review]
LGTM, @yanivagman if you want to take a look before merge

**Comment by AlonZivony** [comment]
> LGTM, @yanivagman if you want to take a look before merge

@rafaeldtinoco already approved, so I don't think we need @yanivagman for it too...

**Review by yanivagman — APPROVED** [review]
LGTM

**Pull request merged**
